operator/v1/types_ingress.go

```diff
@@ -446,6 +452,50 @@ type EndpointPublishingStrategy struct {
 	NodePort *NodePortStrategy `json:"nodePort,omitempty"`
 }
 
+// ClientCertificatePolicy describes the policy for client certificates.
+// +kubebuilder:validation:Enum="";Required;Optional
```

Comment:
Is the `""` necessary here?

Reply:
Hm, I'll need to check what the implications are of removing `""` as an allowed value when `spec.clientTLS` is optional and `spec.clientTLS.clientCertificatePolicy` (which is of this type) is required.

Reply:
`""` turned out to be totes unnecessary, so I've removed it.  Thanks for pointing that out!

Reply:
Cool! Thanks for checking (I wasn't using the "" value in the ExternalDNS operator API).

Reply:
In my testing today, the operator is logging the following:
```
2021-07-13T20:32:51.028Z	ERROR	operator.init	wait/wait.go:155	failed to ensure default ingresscontroller	{"error": "IngressController.operator.openshift.io \"default\" is invalid: spec.clientTLS.clientCertificatePolicy: Unsupported value: \"\": supported values: \"Required\", \"Optional\""}
```
In my earlier testing, I used the CRD without `"";`, but I used the old generated client code, so I can only assume that the `"";` is necessary for the generated client code to work properly.  Alternatively, I can make `ClientTLS` in `IngressControllerSpec` a pointer field; not sure which is less bad.

Reply:
Hm, adding back `"";` doesn't affect the client code.